paimon-flink/paimon-flink-common/src/test/java/org/apache/paimon/flink/source/operator/OperatorSourceTest.java

```diff
@@ -299,6 +300,9 @@ public void emitLatencyMarker(LatencyMarker latencyMarker) {}
 
                     @Override
                     public void emitRecordAttributes(RecordAttributes recordAttributes) {}
+
+                    // @Override is skipped for compatibility with Flink 1.x.
```

Comment:
How is this handle for Spark3 and Spark4?

Reply:
In #4325, spark mainly uses shim package or object to resolve compatibility issues.

If this comment seems improper, we could choose to add implementations for PushingAsyncDataInput.DataOutput with the same name in both paimon-flink1-common and paimon-flink2-common, and use that here and in LocalMergeOperatorTest.

Reply:
Thanks for the explanation! I am good with the comments.